base/libgit2/repository.jl

```diff
@@ -34,6 +34,39 @@ function cleanup(r::GitRepo)
 end
 
 """
+    LibGit2.discover(
+        start_path::AbstractString = pwd();
+        ceiling::Union{AbstractString,Vector} = "",
+        across_fs::Bool = false,
+    )
+
+Look for a git repository starting with `start_path` (by default the current
+working directory), walk across parent directories. Lookup ends when the
+repository is found, or when reaching a directory referenced in `ceiling` or
+when the filesystem changes (in case `across_fs` is `false`). The `ceiling`
+keyword argument can be passed as an array of path strings or as a single string
+which is interpreted as a `:`-separated list of paths (on Windows the path
+separator is `;` instead of `:`). No ceiling directory is the default.
+"""
+function discover(
+    start_path::AbstractString = pwd();
+    ceiling::Union{AbstractString,Vector} = "",
+    across_fs::Bool = false,
+)
+    sep = @static is_windows() ? ";" : ":"
+    ceil = ceiling isa AbstractString ? ceiling :
```

Comment:
You could do this instead by having something like

```julia
function discover(
     start_path::AbstractString = pwd();
     ceiling::Vector{AbstractString} = [""],
     across_fs::Bool = false)
...
end

discover(start_path::AbstractString = pwd(); ceiling::AbstractString = "", across_fs::Bool = false) = discover(start_path, ceiling=[ceiling], across_fs)
```
? Idk if that makes sense